core/src/test/scala/integration/kafka/api/PlaintextConsumerSubscriptionTest.scala

```diff
@@ -178,6 +178,49 @@ class PlaintextConsumerSubscriptionTest extends AbstractConsumerTest {
     assertEquals(0, consumer.assignment().size)
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersConsumerGroupProtocolOnly"))
+  def testRe2JPatternSubscription(quorum: String, groupProtocol: String): Unit = {
+    val topic1 = "tblablac" // matches subscribed pattern
+    createTopic(topic1, 2, brokerCount)
+
+    val topic2 = "tblablak" // does not match subscribed pattern
+    createTopic(topic2, 2, brokerCount)
+
+    val topic3 = "tblab1" // does not match subscribed pattern
+    createTopic(topic3, 2, brokerCount)
+
+    val consumer = createConsumer()
+    assertEquals(0, consumer.assignment().size)
+
+    val pattern = new SubscriptionPattern("t.*c")
+    consumer.subscribe(pattern, new TestConsumerReassignmentListener)
+
+    val assignment = Set(
+      new TopicPartition(topic, 0),
+      new TopicPartition(topic, 1),
+      new TopicPartition(topic1, 0),
+      new TopicPartition(topic1, 1))
+    awaitAssignment(consumer, assignment)
+    consumer.unsubscribe()
```

Comment:
Does it support to subscribe another pattern after unsubscribe? If yes, could you please include such test case?

Reply:
Trying to subscribe after unsubscribe currently fails just because it requires this GC patch still in-flight https://github.com/apache/kafka/pull/17976

Should we move on with this PR and I'll extend and add more tests for subsequent patter subscription after that fix? (all other comments addressed). As you prefer @chia7712 / @dajac

Reply:
I think that we can add more tests afterwards as long as we do it :)